pkgs/top-level/haskell-packages.nix

```diff
@@ -174,18 +209,39 @@ in {
 
     # The integer-simple attribute set contains package sets for all the GHC compilers
     # using integer-simple instead of integer-gmp.
-    integer-simple = let
-      integerSimpleGhcNames = pkgs.lib.filter
-        (name: ! builtins.elem name integerSimpleExcludes)
-        (pkgs.lib.attrNames packages);
-    in pkgs.lib.genAttrs integerSimpleGhcNames (name: packages."${name}".override {
+    integer-simple = pkgs.lib.genAttrs integerSimpleGhcNames (name: packages."${name}".override {
       ghc = bh.compiler.integer-simple."${name}";
       buildHaskellPackages = bh.packages.integer-simple."${name}";
       overrides = _self : _super : {
+        # Usually, integer-simple is a normal package that you can build.
+        # In this package set, it is a ghc-bundled library, so we
+        # set it to `null` (ghc-bundled libraries are always `null`
+        # in `haskellPackages` package sets).
         integer-simple = null;
+        # Get rid of `integer-gmp` as a safety measure to notice
+        # when anything hard-depends on it.
         integer-gmp = null;
       };
     });
 
+    # The dwarf attribute set contains package sets for all the GHC compilers
+    # with all packages compiled with `-g` debugging info.
+    dwarf = pkgs.lib.genAttrs normalGhcCompilers (name: packages."${name}".override {
```

Comment:
@nh2, I just realized that with the current construction it is not possible to add overrides to the `dwarf` package set. This seems like a major limitation that I will inevitably run into.

Reply:
Hmm, anybody have an idea what can be done about this?

Wouldn't that problem exist for all other package sets as well?

Reply:
Indeed, I believe the `integer-simple` set is already broken in this same way.